python/vowpalwabbit/pyvw.py

```diff
@@ -366,19 +372,34 @@ def learn(self, ec):
             ec = self.parse(ec)
             new_example = True
 
+        elif isinstance(ec, list):
+            if self._is_multiline():
+                ec = self.parse(ec)
+            else:
+                ec = list(map(self.parse, ec))
```

Comment:
I think this should be a type error as per my first comment about lists of strings in single-line mode